app/src/main/java/com/benliandamit/tictactoe/ui/MainActivity.kt

```diff
@@ -0,0 +1,94 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.benliandamit.tictactoe.ui.screens.AppBar
+import com.benliandamit.tictactoe.ui.screens.ButtonGrid
+import com.benliandamit.tictactoe.ui.screens.ResetButton
+import com.benliandamit.tictactoe.ui.ui.theme.TicTacToeTheme
+import com.benliandamit.tictactoe.util.StateEnum
+
+class MainActivity : ComponentActivity() {
+    private val mainViewModel by viewModels<MainViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContent {
+            TicTacToeTheme {
+                Scaffold(modifier = Modifier.fillMaxSize(),
+                    topBar = { AppBar() }) { innerPadding ->
+                    Surface(
+                        color = MaterialTheme.colorScheme.background,
+                        modifier = Modifier.fillMaxSize()
+                    ) {
+                        Column(
+                            modifier = Modifier.fillMaxSize(),
+                            verticalArrangement = Arrangement.SpaceEvenly,
+                            horizontalAlignment = Alignment.CenterHorizontally
+                        ) {
+                            ButtonGrid(
+                                board = mainViewModel.board,
+                                onclick = mainViewModel::play
+                            )
+
+
+                            if (mainViewModel.state != StateEnum.CONTINUE) {
+                                val message = when (mainViewModel.state) {
+                                    StateEnum.TIE -> "It's a tie!"
+                                    StateEnum.X -> "Player X wins!"
+                                    StateEnum.O -> "Player O wins!"
+                                    else -> ""
+                                }
+
+                                Box {
+                                    Text(
+                                        text = message,
+                                        style = MaterialTheme.typography.bodyLarge
+                                    )
+                                }
+                            } else {
+                                Text(
+                                    text = "Player ${mainViewModel.player}'s turn",
+                                    style = MaterialTheme.typography.bodyLarge
+                                )
+                            }
+
+                            ResetButton(onClick = mainViewModel::reset)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+}
+
+@Preview(showBackground = true)
+@Composable
+fun DefaultPreview() {
+    TicTacToeTheme {
```

Comment:
if this is the test we talked about you should delete this

Reply:
It's not much of a test. It is best practice to have a "Preview" component so you can see the UI updates live without the need to run the entire app on the phone